json/src/main/java/io/airlift/json/JsonCodec.java

```diff
@@ -158,6 +162,30 @@ public String toJson(T instance)
     }
 
     /**
+     * Converts the specified instance to json string with a maximum length limit.
+     *
+     * @param instance the instance to convert to json
+     * @param lengthLimit the maximum length of the serialized string
+     * @return json string
+     * @throws IllegalArgumentException if the specified instance can not be converted to json or if length limit is exceeded
+     */
+    public String toJsonWithLengthLimit(T instance, int lengthLimit)
+            throws JsonLengthLimitExceededException
+    {
+        try (StringWriter stringWriter = new StringWriter();
+                LengthLimitedWriter lengthLimitedWriter = new LengthLimitedWriter(stringWriter, lengthLimit)) {
+            mapper.writeValue(lengthLimitedWriter, instance);
+            return stringWriter.getBuffer().toString();
+        }
+        catch (LengthLimitedWriter.LengthLimitExceededException e) {
+            throw new JsonLengthLimitExceededException(String.format("%s could not be converted to json string with length limit %d", instance.getClass().getName(), lengthLimit), e);
```

Comment:
I'm not sure about the usefulness of having `getClass().getName()` here. I've not seen this before.